client/rest-high-level/src/main/java/org/elasticsearch/client/transform/transforms/latest/LatestDocConfig.java

```diff
@@ -0,0 +1,134 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.client.transform.transforms.latest;
+
+import org.elasticsearch.common.ParseField;
+import org.elasticsearch.common.xcontent.ConstructingObjectParser;
+import org.elasticsearch.common.xcontent.ObjectParser;
+import org.elasticsearch.common.xcontent.ToXContentObject;
+import org.elasticsearch.common.xcontent.XContentBuilder;
+import org.elasticsearch.common.xcontent.XContentParser;
+import org.elasticsearch.search.sort.FieldSortBuilder;
+import org.elasticsearch.search.sort.SortBuilder;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Objects;
+
+import static org.elasticsearch.common.xcontent.ConstructingObjectParser.constructorArg;
+
+/**
+ * Class describing how to compute latest doc for every unique key
+ */
+public class LatestDocConfig implements ToXContentObject {
+
+    private static final ParseField UNIQUE_KEY = new ParseField("unique_key");
+    private static final ParseField SORT = new ParseField("sort");
+
+    private final List<String> uniqueKey;
+    private final List<SortBuilder<?>> sort;
+
+    @SuppressWarnings("unchecked")
+    private static final ConstructingObjectParser<LatestDocConfig, Void> PARSER =
+        new ConstructingObjectParser<>(
+            "latest_doc_config", true, args -> new LatestDocConfig((List<String>) args[0], (List<SortBuilder<?>>) args[1]));
+
+    static {
+        PARSER.declareStringArray(constructorArg(), UNIQUE_KEY);
```

Comment:
the more I think about it, the more I think we should take option 3 (no setting, default true).

Reply:
Agree.
Option 3 is the simplest one and gives sensible default.
In case users need more customization, we can always introduce per-field setting later.